pkg/crc/machine/vsock.go

```diff
@@ -146,3 +139,39 @@ func getSSHTunnelURI() string {
 	}
 	return u.String()
 }
+
+func HTTPAndHTTPS() []types.ExposeRequest {
+	hPort, hsPort := httpPort, httpsPort
+	if isPortInUse(hPort) {
+		logging.Warnf("Port %s is already in use. Going to use %s", httpPort, alternateHTTPPort)
+		hPort = alternateHTTPPort
+	}
+
+	if isPortInUse(hsPort) {
+		logging.Warnf("Port %s is already in use. Going to use %s", httpsPort, alternateHTTPSPort)
+		hsPort = alternateHTTPSPort
+	}
+
+	return []types.ExposeRequest{
+		{
+			Protocol: "tcp",
+			Local:    fmt.Sprintf(":%s", hsPort),
+			Remote:   net.JoinHostPort(virtualMachineIP, httpsPort),
+		},
+		{
+			Protocol: "tcp",
+			Local:    fmt.Sprintf(":%s", hPort),
+			Remote:   net.JoinHostPort(virtualMachineIP, httpPort),
+		},
+	}
+}
+
+func isPortInUse(port string) bool {
+	ln, err := net.Listen("tcp", ":"+port)
```

Comment:
This could use `fmt.Sprintf` for consistency with the code before it

Reply:
Fwiw, listening on port 80/443 is a privileged operation on some OS, I don't know if `isPortInUse` could fail because of this.

Reply:
> Fwiw, listening on port 80/443 is a privileged operation on some OS, I don't know if `isPortInUse` could fail because of this.

@cfergeau I checked with linux/mac running a python simple http server on port 80 and this is identified by this function.